### PR TITLE
Add case insensitive user attribute keys configs for LDAP

### PR DESCRIPTION
**Pull request by jasonodonnell**
Vault relies on [case insensitive user attribute keys](https://github.com/hashicorp/vault-enterprise/blob/main/sdk/helper/ldaputil/config.go#L302-L304) for mapping user attributes to entity alias metadata. This adds configs to allow the callers to specify how the attribute keys should be returned.

### CODE REVIEW

**Review by fairclothjm — APPROVED** [review]

File: ldap/options.go [hunk 1]
```diff
@@ -75,6 +76,18 @@ func WithUserAttributes() Option {
 	}
 }
 
+// WithLowerUserAttributeKeys returns a User Attribute map where the keys
+// are all cast to lower case. This is neccessary for some clients, such as Vault,
```
Comment (jimlambrt) [inline ldap/options.go:80]:
```suggestion
// are all cast to lower case. This is necessary for some clients, such as Vault,
```

File: ldap/client.go [hunk 2]
```diff
@@ -273,7 +273,11 @@ func (c *Client) Authenticate(ctx context.Context, username, password string, op
 			return nil, fmt.Errorf("%s: failed to get user attributes: %w", op, err)
 		}
 		for _, a := range attrs {
-			userAttrs[a.Name] = a.Vals
+			name := a.Name
```
Comment (jimlambrt) [inline ldap/client.go:276]: Mind adding WithLowerUserAttributeKeys to the list of supported optins in the
func's godoc?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
looks good overall.  couple minor things.

File: ldap/client.go (hunk 2 above)
Comment (jasonodonnell) [reply ldap/client.go:276]: Good call, done

**Review by jimlambrt — APPROVED** [review]
ty!

**Pull request merged**